core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java

```diff
@@ -152,40 +148,37 @@ public void stop() {
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
 		String forward="";
 		PipeForward pipeForward = null;
-
-		if (transformerPool!=null) {
-			ParameterList parameterList = null;
+		if(StringUtils.isNotEmpty(getForwardNameSessionKey())) {
+			try {
+				forward = session.getMessage(getForwardNameSessionKey()).asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, getLogPrefix(session)+"cannot open stream", e);
+			}
+		} else if(!(StringUtils.isEmpty(getXpathExpression()) && StringUtils.isEmpty(getStyleSheetName())) || StringUtils.isEmpty(getSessionKey())) {
 			try {
 				Map<String,Object> parametervalues = null;
-				parameterList =  getParameterList();
-				if (parameterList!=null) {
-					message.preserve();
```

Comment:
Can the preserve be removed?

Reply:
In this case, it cannot so put it back :)